### PR TITLE
CEDS-2006/CEDS-2007 Ensure Logging & Errors handled appropriately

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/config/ErrorHandler.scala [hunk 1]
```diff
@@ -17,14 +17,28 @@
...
-import play.api.mvc.Request
+import play.api.mvc.{Request, RequestHeader, Result, Results}
 import play.twirl.api.Html
 import uk.gov.hmrc.play.bootstrap.http.FrontendErrorHandler
 import views.html.error
 
 @Singleton
 class ErrorHandler @Inject()(val messagesApi: MessagesApi, errorTemplate: error) extends FrontendErrorHandler {
+
+  private val logger = Logger(classOf[ErrorHandler])
```
Comment (PatrykRudnicki) [inline app/config/ErrorHandler.scala]: `this.getClass()` will return the same and you will not have a risk that you'll change this class to something different, but this approach is correct too

File: app/config/ErrorHandler.scala [hunk 2]
```diff
@@ -17,14 +17,28 @@
...
+
   override def standardErrorTemplate(pageTitle: String, heading: String, message: String)(implicit request: Request[_]): Html =
     errorTemplate(pageTitle, heading, message)
+
+  override def resolveError(rh: RequestHeader, ex: Throwable): Result = ex match {
```
Comment (PatrykRudnicki) [inline app/config/ErrorHandler.scala]: I'll put whole name, not shortcuts for `requestHeader` and `exception`

File: app/connectors/CustomsDeclareExportsMovementsConnector.scala [hunk 3]
```diff
@@ -101,6 +95,12 @@ class CustomsDeclareExportsMovementsConnector @Inject()(appConfig: AppConfig, ht
       }
 
   private def providerIdQueryParam(providerId: String): Seq[(String, String)] = Seq("providerId" -> providerId)
+
+  private def logSuccessfulExchange[T](`type`: String, payload: T)(implicit fmt: Format[T]): Unit =
+    logger.debug(`type` + "\n" + Json.toJson(payload))
+
+  private def logFailedExchange(`type`: String, exception: Throwable): Unit =
+    logger.error(`type` + " failed", exception)
```
Comment (SteveSugden) [inline app/connectors/CustomsDeclareExportsMovementsConnector.scala]: Just curious - why error not warn?

Comment (MaciejRewera) [reply app/connectors/CustomsDeclareExportsMovementsConnector.scala]: If I remember correctly, `error` level logs are treated very seriously and might even cause us being called in the middle of the night (?)
I'd rather use `warn` in that case.

Comment (ed0906) [reply app/connectors/CustomsDeclareExportsMovementsConnector.scala]: This happens when the back end returns a non-2xx, in which case we probaly want to know why

Comment (ed0906) [reply app/connectors/CustomsDeclareExportsMovementsConnector.scala]: @MaciejRewera that's only if enough error level logs are raised to reach the PagerDuty threshold

Comment (SteveSugden) [reply app/connectors/CustomsDeclareExportsMovementsConnector.scala]: As long as I don't have to get out of bed, I approve.

Comment (bambuchaAdm) [reply app/connectors/CustomsDeclareExportsMovementsConnector.scala]: On the other hand is this won't be logged on error-handler on top level ?

**Pull request merged**